### ISSUE TITLE
list index out of range for some docstring in 4.0 (fixed in 5.0.dev/master)

### ISSUE THREAD

**Issue by pombreda** [mitsuhiko/click#369]
with this (note the empty line after the line `hello`):
```python
import click

@click.command()
def cli():
    """\b
Print Hello:
    hello

    """
    click.echo('hello')
```
I get this:
```
test --help

Traceback (most recent call last):
  File "/home/me/bin/inout", line 9, in <module>
    load_entry_point('click-example-inout==0.1', 'console_scripts', 'inout')()
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 643, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 560, in make_context
    self.parse_args(ctx, args)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 822, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 1326, in handle_parse_result
    self.callback, ctx, self, value)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 54, in invoke_param_callback
    return callback(ctx, param, value)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 751, in show_help
    echo(ctx.get_help(), color=ctx.color)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 416, in get_help
    return self.command.get_help(self)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 772, in get_help
    self.format_help(ctx, formatter)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 786, in format_help
    self.format_help_text(ctx, formatter)
  File "/home/me/local/lib/python2.7/site-packages/click/core.py", line 795, in format_help_text
    formatter.write_text(self.help)
  File "/home/me/local/lib/python2.7/site-packages/click/formatting.py", line 165, in write_text
    preserve_paragraphs=True))
  File "/home/me/local/lib/python2.7/site-packages/click/formatting.py", line 90, in wrap_text
    post_wrap_indent))
  File "/home/me/local/lib/python2.7/site-packages/click/formatting.py", line 23, in add_subsequent_indent
    lines = [lines[0]] + [subsequent_indent + line for line in lines[1:]]
IndexError: list index out of range
```

Removing the extra empty line fixes the issue.

The issue does not exist in 5.0/master  but I though it worth to report if someone stumbles on it.

**Comment by pombreda** [comment]
Fixed in 5.0/master.

**Comment by pombreda** [comment]
For reference, this commit fixed it: https://github.com/mitsuhiko/click/commit/4e519da58ac79fbd97eb86eb7717603d77eec379
